backend/src/api_parsers/dblp_parser.py

```diff
@@ -0,0 +1,106 @@
+from requests.exceptions import HTTPError
+from src.api_handlers.dblp.handler import DblpHandler
+from src.api_handlers.core.exceptions import QuotaExceededException
+from src.api_parsers.models import Source, Publication, Author
+from src.api_parsers.exceptions import (
+    NoAuthorsException,
+    NoAffiliationException,
+    DBLPQuotaExceededException,
+)
+
+
+class DBLPParser:
+    def __init__(self, keywords: str):
+        self.keywords = keywords
+        self.handler = DblpHandler()
+
+    def get_authors(self) -> list[Author]:
+        authors: list[Author] = []
+        try:
+            pub_response = self.handler.get_publications(self.keywords)
+            for page in pub_response:
+                page_authors = self._parse_publications_page(page)
+                authors.extend(page_authors)
+        except DBLPQuotaExceededException as err:
+            authors.extend(err.authors)
+            return authors
+        except (HTTPError, QuotaExceededException):
+            raise NoAuthorsException(self.keywords)
+        return authors
+
+    def _parse_publications_page(self, page: dict) -> list[Author]:
+        authors: list[Author] = []
+        hits = page["result"]["hits"]["hit"]
+        for hit in hits:
+            try:
+                authors.extend(self._parse_hit_dict(hit))
+            except DBLPQuotaExceededException as err:
+                authors.extend(err.authors)
+                raise DBLPQuotaExceededException(authors)
```

Comment:
I'm confused what is going on here, why are we extending the list and then raising the exception? We wont be able to access this list anyways

Reply:
the list is propagated upwards via the exception, so that we're still able to access its (incomplete) contents knowing that there was an error. I could have used a flag for that as well though

Reply:
Exception don't need to carry atributes like that, use logging for debugging purposes (displaying authors)

Reply:
this attribute is not used only for debugging though. the authors are still returned as result, and used when the newly raised exception is caught by the calling function. this way it's indicated that searching for authors stopped because quota was exceeded, not because there were no more results. elsewise there would occur that loop of exceptions we were talking about last week. (see lines 24-29)

Reply:
alright changed this approach to storing list of authors as class attribute. this way no attribute is carried by a chain of exceptions and functions.